server/src/main/java/org/elasticsearch/cluster/ClusterFeatures.java

```diff
@@ -79,28 +80,61 @@ public Map<String, Set<String>> nodeFeatures() {
         return nodeFeatures;
     }
 
-    /**
-     * The features in all nodes in the cluster.
-     * <p>
-     * NOTE: This should not be used directly.
-     * Please use {@link org.elasticsearch.features.FeatureService#clusterHasFeature} instead.
-     */
-    public Set<String> allNodeFeatures() {
+    private Set<String> allNodeFeatures() {
         if (allNodeFeatures == null) {
             allNodeFeatures = Set.copyOf(calculateAllNodeFeatures(nodeFeatures.values()));
         }
         return allNodeFeatures;
     }
 
+    /**
+     * Returns {@code true} if {@code node} can have assumed features.
+     * @see org.elasticsearch.env.BuildVersion#canRemoveAssumedFeatures
+     */
+    public static boolean featuresCanBeAssumedForNode(DiscoveryNode node) {
+        return node.getBuildVersion().canRemoveAssumedFeatures();
```

Comment:
Stumbled over `remove` in `canRemoveAssumedFeatures`, that was confusing to me and I kept looking for where we remove those.  Maybe turning it around makes it easier to understand: `canAssumeRemovedFeatures`

Reply:
That's not quite right though, its not that all features that are missing can be assumed, it's that the features _that this ES has marked as assumed_ can be taken as assumed in a node with this version, where they are missing

Reply:
Yes, you're right 👍 
If we can find something better than remove, it'd be great, but certainly not worth delaying this further.
`canInferAssumedFeatures`, `canExpectAssumedFeatures`, `canSkipAssumedFeatures`
Or more targeting the versioning side of this: `hasPassedNextCompatiblityBoundary` or similar.